### PR TITLE
commands: require wallet password for lnpay command

### PR DESCRIPTION
**Pull request by accumulator**
ref #9236

### CODE REVIEW

**Comment by accumulator** [comment]
might be useful to also require password for e.g. channel closes and reverse swaps

File: electrum/commands.py [hunk 1]
```diff
@@ -1159,8 +1159,10 @@ async def decode_invoice(self, invoice: str):
         invoice = Invoice.from_bech32(invoice)
         return invoice.to_debug_json()
 
-    @command('wnl')
-    async def lnpay(self, invoice, timeout=120, wallet: Abstract_Wallet = None):
+    @command('wnpl')
+    async def lnpay(self, invoice, timeout=120, password=None, wallet: Abstract_Wallet = None):
+        if wallet.has_password():
+            wallet.check_password(password)
```
Comment (SomberNight) [inline electrum/commands.py:1165]: Conceptually this should not be done here I think.
I moved this check to the `@command` decorator in https://github.com/spesmilo/electrum/commit/37d090c621006cc4caa7e0fb9d080b3e6a9da05c.

**Comment by SomberNight** [comment]
> might be useful to also require password for e.g. channel closes and reverse swaps

Well, yes, maybe. But also something to consider is what exactly is the threat-model, because at the end of the day we might as well password-protect every command :P


```
>>> [cn for (cn, c) in electrum.commands.known_commands.items() if c.requires_wallet and not c.requires_password]
```
`['get', 'listunspent', 'freeze', 'unfreeze', 'freeze_utxo', 'unfreeze_utxo', 'ismine', 'getpubkeys', 'getbalance', 'getmpk', 'onchain_history', 'lightning_history', 'setlabel', 'listcontacts', 'getalias', 'searchcontacts', 'listaddresses', 'get_request', 'get_invoice', 'list_requests', 'list_invoices', 'createnewaddress', 'changegaplimit', 'getminacceptablegap', 'getunusedaddress', 'add_request', 'addtransaction', 'delete_request', 'delete_invoice', 'clear_requests', 'clear_invoices', 'is_synchronized', 'removelocaltx', 'get_tx_status', 'add_peer', 'list_peers', 'lnpay', 'nodeid', 'list_channels', 'enable_htlc_settle', 'close_channel', 'request_force_close', 'export_channel_backup', 'import_channel_backup', 'get_channel_ctx', 'get_watchtower_ctn', 'rebalance_channels', 'reverse_swap']`

1. Dangerous commands:
`lnpay, get_channel_ctx`

2. Not too dangerous but still kind of:
`close_channel, request_force_close, export_channel_backup, reverse_swap, rebalance_channels`

3. And then there is also stuff like `getmpk` (privacy) or `changegaplimit` (DOS-potential).

Anyway, I would say let's password-protect the "dangerous" list (1).

Do you think list (2) should be password-protected as well? I am not sure.

File: electrum/commands.py (hunk 1 above)
Comment (accumulator) [reply electrum/commands.py:1165]: agreed

**Comment by accumulator** [comment]
> Do you think list (2) should be password-protected as well? I am not sure.

These are currently inconsistent across GUIs (PIN on android, if enabled. no auth on desktop). 
I'm inclined to add password requirement for (2)

**Comment by SomberNight** [comment]
>  I'm inclined to add password requirement for (2), though only for CLI

Okay, let's do that then.

**Review by SomberNight — APPROVED** [review]
Thanks!

**Pull request merged**